cdap-test/src/main/java/co/cask/cdap/test/AbstractProgramManager.java

```diff
@@ -34,6 +37,19 @@ public AbstractProgramManager(Id.Program programId, ApplicationManager applicati
   }
 
   @Override
+  public T start() {
+    return start(ImmutableMap.<String, String>of());
+  }
+
+  @Override
+  public T start(Map<String, String> arguments) {
+    applicationManager.startProgram(programId, arguments);
+    // this cast is fine as long as the derived classes extend AbstractProgramManager with the
+    // template (<T>) declared as its own class
+    return (T) this;
```

Comment:
I think you don't need to cast if you have `<T extends ProgramManager>` in the `ProgramManager` interface.